src/Microsoft.ML.AutoML/API/BinaryClassificationExperiment.cs

```diff
@@ -166,5 +324,79 @@ private protected override CrossValidationRunDetail<BinaryClassificationMetrics>
+            var trainerName = pipeline.ToString();
+            var parameter = result.TrialSettings.Parameter;
+            var estimator = pipeline.BuildTrainingPipeline(Context, parameter);
+            var modelContainer = new ModelContainer(Context, result.Model);
+            return new RunDetail<BinaryClassificationMetrics>(trainerName, estimator, null, modelContainer, result.BinaryClassificationMetrics, result.Exception);
+        }
+
+        private CrossValidationRunDetail<BinaryClassificationMetrics> ToCrossValidationRunDetail(BinaryClassificationTrialResult result)
+        {
+            var pipeline = result.TrialSettings.Pipeline;
+            var trainerName = pipeline.ToString();
+            var parameter = result.TrialSettings.Parameter;
+            var estimator = pipeline.BuildTrainingPipeline(Context, parameter);
+            var crossValidationResult = result.CrossValidationMetrics.Select(m => new TrainResult<BinaryClassificationMetrics>(new ModelContainer(Context, m.Model), m.Metrics, result.Exception));
+            return new CrossValidationRunDetail<BinaryClassificationMetrics>(trainerName, estimator, null, crossValidationResult);
+        }
+    }
+
+    internal class BinaryClassificationTrialResultMonitor : IMonitor
+    {
+        public BinaryClassificationTrialResultMonitor()
+        {
+            this.RunDetails = new List<BinaryClassificationTrialResult>();
+        }
+
+        public event EventHandler<BinaryClassificationTrialResult> OnTrialCompleted;
+
+        public List<BinaryClassificationTrialResult> RunDetails { get; }
+
+        public BinaryClassificationTrialResult BestRun { get; private set; }
+
+        public void ReportBestTrial(TrialResult result)
+        {
+            if (result is BinaryClassificationTrialResult binaryClassificationResult)
+            {
+                BestRun = binaryClassificationResult;
+            }
+            else
+            {
+                throw new ArgumentException($"result must be of type {typeof(BinaryClassificationTrialResult)}");
+            }
+        }
+
+        public void ReportCompletedTrial(TrialResult result)
+        {
+            if (result is BinaryClassificationTrialResult binaryClassificationResult)
+            {
+                RunDetails.Add(binaryClassificationResult);
+                OnTrialCompleted?.Invoke(this, binaryClassificationResult);
+            }
+            else
+            {
+                throw new ArgumentException($"result must be of type {typeof(BinaryClassificationTrialResult)}");
+            }
+        }
+
+        public void ReportFailTrial(TrialSettings settings, Exception exp)
+        {
+            var result = new BinaryClassificationTrialResult
+            {
+                TrialSettings = settings,
+                Exception = exp,
+            };
+
+            RunDetails.Add(result);
+        }
+
+        public void ReportRunningTrial(TrialSettings setting)
```

Comment:
Do you need this method since it does nothing?

Reply:
This method is to update a running trial and is called in `NotebookMonitor`. So it's better for it to be there

Reply:
👍